pkg/trace/agent/agent.go

```diff
@@ -69,6 +71,11 @@ func NewAgent(ctx context.Context, conf *config.AgentConfig) *Agent {
 	dynConf := sampler.NewDynamicConfig(conf.DefaultEnv)
 	in := make(chan *api.Payload, 1000)
 	statsChan := make(chan pb.StatsPayload, 100)
+	// Instantiate AppSec
+	appsecHandler, err := appsec.New(coreconfig.Datadog)
```

Comment:
A few notes here, I'm adding numbers in case you want to answer each separately:

1. `coreconfig.Datadog` is a global object accessible from any package, why do you need to have it as an argument? I suspect you might've done it for tests, in which case, I'd say to make the public API `appsec.New` and have that call an internal unexported function `new` which takes the config and then use that for tests. No reason to sacrifice "public" API for testability.
2. I think that this code can safely go into the `api.NewHTTPReceiver` function. This seems like the wrong place.

Reply:
Done as stated. I'll create the private `new` function if needed when writing the tests.